google/auth/transport/requests.py

```diff
@@ -281,6 +289,13 @@ def request(
         # thread-safety.
         _credential_refresh_attempt = kwargs.pop("_credential_refresh_attempt", 0)
 
+        # Until all libraries expose explicit timeouts, a rouge None timeout
+        # might sneak in - for example, a default timeout value in one of the
+        # internal classes. Since it's desired to have at least some timeout,
+        # we override any `None` values for now.
+        if timeout is None:
+            timeout = _DEFAULT_TIMEOUT
```

Comment:
This is of course not ideal, but might be a reasonable short- to mid- term fix to stalled requests in higher-level libraries. I am open for feedback of course.

Reply:
This bit is a bit more concerning to me. If a library specifies None, we should likely respect it.

Reply:
Fair enough, I'll purge that commit.